EIPS/eip-6780.md

```diff
@@ -0,0 +1,65 @@
+
+## Motivation
+
+The `SELFDESTRUCT` opcode requires large changes to the state of an account, in particular removing all code and storage. This will not be possible in the future with Verkle trees: Each account will be stored in many different account keys, which will not be obviously connected to the root account.
+
+This EIP implements this change. Applications that only use `SELFDESTRUCT` to retrieve funds will still work. Applications that only use `SELFDESTRUCT` in the same transaction 
+
+## Specification
+
+The behaviour of `SELFDESTRUCT` is changed in the following way:
+
+1. Case where `SELFDESTRUCT` is executed in a transaction that is not the same as the contract calling `SELFDESTRUCT` was created:
+
+ - Does not delete any storage keys or code
+ - Transfer the account balance to the target **and** set account balance to `0.`
+ - Note that no refund is given since [EIP-3529](./eip-3529.md).
+ - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+  
+2. Case where `SELFDESTRUCT` is executed in the same transaction the contract was created
+
+ - `SELFDESTRUCT` continues to behave as originally, i.e. deletes all storage keys and the account itself.
+ - Subsequently, the account will behave like exactly like an empty account, both in the same transaction and in all later ones
+ - Transfer the account balance to the target **and** set account balance to `0.`
+ - Note that no refund is given since [EIP-3529](./eip-3529.md).
+ - Note that the rules of [EIP-2929](./eip-2929.md) regarding `SELFDESTRUCT` remain unchanged.
+ - Note that when verkle tries are implemented on Ethereum, the cleared storage will be marked as having been written before but empty. This leads to no observable differences in EVM execution, but a contract having been created and deleted will lead to different state roots compared to the action not happening.
+
+The `SELFDESTRUCT` opcode remains deprecated as specified in [EIP-6049](./eip-6049.md). Any use in newly deployed contracts is strongly discouraged even if this new behaviour is taken into account, and future changes to the EVM might further reduce the functionality of the opcode.
+
+## Rationale
+
+Getting rid of the `SELFDESTRUCT` opcode has been considered in the past, and there are currently no strong reasons to use it. Disabling it will be a requirement for statelessness. This EIP implements a behaviour that will attempt to leave some known uses of `SELFDESTRUCT` working, while reducing the complexity of the change on EVM implementations that would come from contract versioning (as suggested in [EIP-6190](./eip-6190.md)). A further option that was considered was to just remove storage clearing from `SELFDESTRUCT` in [EIP-6046](./eip-6046.md) while preserving removal of contracts, however this is not safe because existing contracts rely on storage being empty when they are deployed.
```

Comment:
```suggestion
Getting rid of the `SELFDESTRUCT` opcode has been considered in the past, and there are currently no strong reasons to use it. This EIP implements a behavior that will attempt to leave some common uses of `SELFDESTRUCT` working, while reducing the complexity of the change on EVM implementations that would come from contract versioning. A further option that was considered was to just remove storage clearing from `SELFDESTRUCT` in [EIP-6046](./eip-6046.md) while preserving removal of contracts, however this is not safe because existing contracts rely on storage being empty when they are deployed.
```
"Disabling it will be a requirement for statelessness" is false, and EIP-6190 does not define contract versioning.

Also, the phrasing of "This EIP implements a behavior that will attempt to leave some common uses of `SELFDESTRUCT` working, while reducing the complexity of the change on EVM implementations that would come from contract versioning." is clunky, and I suggest you further refine it.